### PR TITLE
#518 - Deploy default cluster-autoscaler after cluster create

### LINKED ISSUE #518: Deploy default cluster-autoscaler after cluster create
Deploy default cluster-autoscaler as a deployment in a posthook after cluster creation on AWS & Azure. In case of GKE setup GKE's own cluster-autoscaler via GKE API. Cluster-autoscaler can be enabled/disabled for each nodepool, when creating updating cluster nodepools. Update will be supported only in case of GKE for now.

### PR DESCRIPTION
**Pull request by sancyx**
Fixes #518

### CODE REVIEW

File: cluster/hooks.go [hunk 1]
```diff
@@ -126,6 +127,66 @@ func InstallIngressControllerPostHook(cluster CommonCluster) {
 	log.Infof("'%s' installed", deploymentName)
 }
 
+//InstallClusterAutoscalerPostHook post hook only for AWS & Azure for now
+func InstallClusterAutoscalerPostHook(cluster CommonCluster) {
+
+	// --- [ Get K8S Config ] --- //
```
Comment (baluchicken) [inline cluster/hooks.go]: Move this to line 150

File: cluster/hooks.go [hunk 2]
```diff
@@ -126,6 +127,66 @@ func InstallIngressControllerPostHook(cluster CommonCluster) {
...
+		return
+	}
+
+	kubeConfig, err := cluster.GetK8sConfig()
+	if err != nil {
+		log.Errorf("Unable to fetch config for posthook: %s", err.Error())
+		return
+	}
+
+	deploymentName := "banzaicloud-stable/cluster-autoscaler"
```
Comment (baluchicken) [inline cluster/hooks.go]: this can be a const

File: cluster/hooks.go [hunk 3]
```diff
@@ -126,6 +127,66 @@ func InstallIngressControllerPostHook(cluster CommonCluster) {
...
+	releaseName := "autoscaler"
+
+	org, err := auth.GetOrganizationById(cluster.GetOrganizationId())
+	if err != nil {
+		log.Errorf("Error during getting organization: %s", err.Error())
+		return
+	}
+
+	var values *autoscalingInfo
+	if cluster.GetType() == constants.Amazon {
```
Comment (baluchicken) [inline cluster/hooks.go]: Minor thing: this can be a switch case

File: cluster/hooks.go [hunk 4]
```diff
@@ -126,6 +127,66 @@ func InstallIngressControllerPostHook(cluster CommonCluster) {
...
 
+//InstallClusterAutoscalerPostHook post hook only for AWS & Azure for now
+func InstallClusterAutoscalerPostHook(cluster CommonCluster) {
+
+	// --- [ Get K8S Config ] --- //
+	log = logger.WithFields(logrus.Fields{"action": "InstallClusterAutoscaler"})
+
+	var nodeGroups []nodeGroup
+
+	if cluster.GetType() == constants.Amazon {
```
Comment (baluchicken) [inline cluster/hooks.go]: Minor thing: this can be a switch case

File: helm/helm.go [hunk 5]
```diff
@@ -198,6 +198,11 @@ func UpgradeDeployment(deploymentName, chartName string, values []byte, reuseVal
 
 //CreateDeployment creates a Helm deployment
```
Comment (baluchicken) [inline helm/helm.go]: Maybe update the doc to say this function only creates a Deployment in a "default" namespace

File: cluster/autoscale.go [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+				MaxSize: nodePool.NodeMaxCount,
+			})
+		}
+	}
+	return nodeGroups
+}
+
+func createAutoscalingForAmazon(cluster CommonCluster, groups []nodeGroup) *autoscalingInfo {
+	return &autoscalingInfo{
+		CloudProvider:     "aws",
```
Comment (stoader) [inline cluster/autoscale.go]: make it a const

File: cluster/autoscale.go [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+		AutoscalingGroups: groups,
+		ExtraArgs: map[string]string{
+			"v":        "5",
```
Comment (stoader) [inline cluster/autoscale.go]: make it a const

File: cluster/autoscale.go [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+			"expander": "least-waste",
```
Comment (stoader) [inline cluster/autoscale.go]: make it a conts

**Pull request merged**